cli/tests/extensions.rs

```diff
@@ -0,0 +1,298 @@
+            .collect::<Vec<_>>()
+            .len()
+            > 1
+    );
+
+    Command::cargo_bin("phylum")
+        .unwrap()
+        .env("XDG_DATA_HOME", &tmp_dir)
+        .arg("extension")
+        .arg("remove")
+        .arg("sample-extension")
+        .assert()
+        .success();
+
+    assert!(!extension_path.exists());
+}
+
+// When a user runs phylum extension or phylum extension list a list of
+// currently installed extensions, their versions and a short one sentence blurb
+// on what the extension does should be shown in a table format.
+#[test]
+fn extension_list_should_emit_output() {
+    let tmp_dir = TmpDir::new();
+
+    // Output that no extension is installed when that is the case.
+    let cmd = Command::cargo_bin("phylum")
+        .unwrap()
+        .env("XDG_DATA_HOME", &tmp_dir)
+        .arg("extension")
+        .arg("list")
+        .assert();
+
+    let output = std::str::from_utf8(&cmd.get_output().stdout).unwrap();
+    let re = Regex::new(r#"No extension"#).unwrap();
+
+    assert!(output.lines().find(|m| re.is_match(m)).is_some());
+
+    // Install one extension
+    Command::cargo_bin("phylum")
+        .unwrap()
+        .env("XDG_DATA_HOME", &tmp_dir)
+        .arg("extension")
+        .arg("add")
+        .arg(fixtures_path().join("sample-extension"))
+        .assert();
+
+    // Output name and description of the extension when one is installed
+    let cmd = Command::cargo_bin("phylum")
+        .unwrap()
+        .env("XDG_DATA_HOME", &tmp_dir)
+        .arg("extension")
+        .arg("list")
+        .assert();
+
+    let output = std::str::from_utf8(&cmd.get_output().stdout).unwrap();
+    let re = Regex::new(r#"^sample-extension\s+This extension does a thing"#).unwrap();
+
+    assert!(output.lines().find(|m| re.is_match(m)).is_some());
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Miscellaneous tests
+////////////////////////////////////////////////////////////////////////////////
+
+#[test]
+fn valid_extension_is_loaded_correctly() {
+    let ext = Extension::try_from(fixtures_path().join("sample-extension")).unwrap();
+
+    assert_eq!(ext.name(), "sample-extension");
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Utilities
+////////////////////////////////////////////////////////////////////////////////
+
+fn project_root() -> PathBuf {
+    Path::new(&env!("CARGO_MANIFEST_DIR"))
+        .ancestors()
+        .nth(1)
+        .unwrap()
+        .to_path_buf()
+}
+
+fn fixtures_path() -> PathBuf {
+    project_root()
+        .join("cli")
+        .join("tests")
+        .join("fixtures")
+        .join("extensions")
+}
+
+struct TmpDir(PathBuf);
+
+impl TmpDir {
+    fn new() -> Self {
+        let dir_name: String = rand::thread_rng()
+            .sample_iter(&rand::distributions::Alphanumeric)
+            .take(16)
+            .map(char::from)
+            .collect();
```

Comment:
It'd be nice to use something descriptive in the name (e.g., `phylum-test-...` where `...` is the 16 random alphanumeric characters)